cpp/src/arrow/python/python_to_arrow.cc

```diff
@@ -868,6 +868,8 @@ class DecimalConverter
   using BASE =
       TypedConverter<arrow::Decimal128Type, DecimalConverter<null_coding>, null_coding>;
 
+  DecimalConverter() : BASE(), decimal_type_(nullptr) {}
```

Comment:
I don't think you even need that. A `shared_ptr` is always initialized to null by default (unline a raw pointer which is uninitialized).

Reply:
I see. I've removed this.